uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Mail/MessageTest.php

```diff
@@ -0,0 +1,208 @@
+      $this->assertContains($value['email'], [
+        'staffmember@dummysite.com', 'approver1@dummysite.com', 'approver2@dummysite.com'
+      ]);
+
+      $this->assertContains($value['api.Contact.get']['values'][0]['display_name'], [
+        'Approver1 Approver1', 'Staff1 Staff1', 'Approver2 Approver2'
+      ]);
+    }
+  }
+
+  public function testGetRecipientEmailsReturnsEmailsForContactAndLeaveApproverOnlyWhenLeaveContactHasLeaveApproverAndThereIsDefaultLeaveApproverForTheAbsenceType() {
+    $manager1 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager1', 'last_name' => 'Manager1'], 'manager1@dummysite.com'
+    );
+
+    $this->setLeaveApproverRelationshipTypes(['has Leaves Approved By']);
+
+    // Set manager1 to be leave aprovers for the leave contact
+    $this->setContactAsLeaveApproverOf($manager1, $this->leaveContact, null, null, true, 'has Leaves Approved By');
+
+    $defaultApprover1 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Approver1', 'last_name' => 'Approver1'], 'approver1@dummysite.com'
+    );
+
+    $absenceType = 1;
+    //add a default leave approvers for the absence type
+    NotificationReceiver::addReceiversToAbsenceType($absenceType, [$defaultApprover1['id']]);
+
+    $leaveRequest = LeaveRequest::create([
+      'type_id' => $absenceType,
+      'contact_id' => $this->leaveContact['id'],
+      'status_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('tomorrow'),
+      'from_date_type' => 1,
+      'to_date' => CRM_Utils_Date::processDate('tomorrow'),
+      'to_date_type' => 1,
+      'toil_to_accrue' => 2,
+      'toil_duration' => 120,
+      'request_type' => LeaveRequest::REQUEST_TYPE_TOIL
+    ], false);
+
+    $message = new Message($leaveRequest, $this->leaveRequestTemplateFactory);
+    $recipientEmails = $message->getRecipientEmails($leaveRequest);
+
+    $this->assertCount(2, $recipientEmails);
+
+    //since there are leave approvers for the leave contact, no mails will be sent to the default approvers
+    //for the absence type
+    foreach($recipientEmails as $value) {
+      $this->assertContains($value['email'], [
+        'staffmember@dummysite.com', 'manager1@dummysite.com'
+      ]);
+
+      $this->assertContains($value['api.Contact.get']['values'][0]['display_name'], [
+        'Manager1 Manager1', 'Staff1 Staff1'
+      ]);
+    }
+  }
+
+  public function testGetLeaveContact() {
+    $leaveRequest = new LeaveRequest();
+    $leaveRequest->contact_id = 2;
+    $message = new Message($leaveRequest, $this->leaveRequestTemplateFactory);
+    $this->assertEquals($leaveRequest->contact_id, $message->getLeaveContactID());
+  }
+
+  public function testGetTemplateParameters() {
+    $notificationTemplateFactory = $this->createRequestNotificationTemplateFactoryMock();
+
+    $leaveRequest = new LeaveRequest();
+    $message = new Message($leaveRequest, $notificationTemplateFactory);
+    $message->getTemplateParameters();
```

Comment:
Well its actually asserting that the getTemplateParameters of the Template class is at least called. 
All the method does is this
`  public function getTemplateParameters() {
    return $this->getTemplate()->getTemplateParameters($this->leaveRequest);
  }`
and since the `getTemplateParameters` had already been tested, I only wanted to make sure that the message class calls the method. 
What do you think?